sdk/servicebus/service-bus/test/internal/connectionString.spec.ts

```diff
@@ -0,0 +1,109 @@
+  const expectedEntityPath = "my-entityPath";
+  const expectedSharedAccessSignature = "my-shared-access-signature";
+  const expectedSharedKey = "my-shared-key";
+  const expectedSharedKeyName = "my-shared-key-name";
+  const dummyErrorMsg = "You should not be seeing me";
+
+  it("Extracts Service Bus properties with only Endpoint", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+  });
+
+  it("Extracts Service Bus properties with Endpoint & EntityPath", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};EntityPath=${expectedEntityPath}`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+    assert.equal(connectionStringProperties.entityPath, expectedEntityPath);
+  });
+
+  it("Extracts Service Bus properties with Endpoint & SAS", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};SharedAccessSignature=${expectedSharedAccessSignature}`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+    assert.equal(connectionStringProperties.sharedAccessSignature, expectedSharedAccessSignature);
+  });
+
+  it("Extracts Service Bus properties with Endpoint & SharedKey", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};SharedAccessKey=${expectedSharedKey};SharedAccessKeyName=${expectedSharedKeyName}`;
+    const connectionStringProperties = parseServiceBusConnectionString(connectionString);
+    assert.equal(connectionStringProperties.endpoint, expectedEndpoint);
+    assert.equal(connectionStringProperties.fullyQualifiedNamespace, expectedNamespace);
+    assert.equal(connectionStringProperties.sharedAccessKey, expectedSharedKey);
+    assert.equal(connectionStringProperties.sharedAccessKeyName, expectedSharedKeyName);
+  });
+
+  it("Throws when no Endpoint", () => {
+    const connectionString = `EntityPath=${expectedEntityPath};SharedAccessSignature=${expectedSharedAccessSignature}`;
+    try {
+      parseServiceBusConnectionString(connectionString);
+      throw new Error(dummyErrorMsg);
+    } catch (error) {
+      assert.notEqual(error.message, dummyErrorMsg);
+    }
+  });
+
+  it("Throws when SharedKey with no SharedKeyName", () => {
+    const connectionString = `Endpoint=${expectedEndpoint};SharedAccessKey=${expectedSharedKey}`;
+    try {
+      parseServiceBusConnectionString(connectionString);
```

Comment:
I think there actually is an 'assert.throws' that you could use if you wanted to reduce boilerplate in the future.

Reply:
I like that. Done in https://github.com/Azure/azure-sdk-for-js/pull/11949/commits/548d83ef79c58ad8c37dca021acbeb7b124a4f6c